consensus/common_test.go

```diff
@@ -37,6 +37,23 @@ func ensureDir(dir string, mode os.FileMode) {
 	}
 }
 
+type NodeConfig struct {
+	cfg.Config `mapstructure:",squash"`
+	P2P        *p2p.Config   `mapstructure:"p2p"`
+	Mempool    *mempl.Config `mapstructure:"mempool"`
+	Consensus  *Config       `mapstructure:"consensus"`
+}
+
+// TODO: This is the same as NodeConfig. Should we move the configs to types (?)
```

Comment:
An alternative is to give up control of eg the p2p config details in the consensus tests from the test toml file. Eg. here we only unmarshal the piece we need: https://github.com/tendermint/tendermint/blob/config_structs/mempool/mempool_test.go#L14

We could do that also in consensus test and just use p2p.NewDefaultConfig where we need a p2p config ...